woocommerce/google-listings-and-ads
# GLA Request review - Add Onboarding notice

### Changes proposed in this Pull Request:

This PR adds the Onboarding Status to the Account Status as well as other minor refactoring. 

- The onboarding status will be there when the status of any of the program regions is Onboarding or if any of the global states is `NO_OFFERS_UPLOADED`

Its present in the frontend like this
<img width="1164" alt="Screenshot 2022-04-28 at 17 26 42" src="https://user-images.githubusercontent.com/5908855/165762817-5fa4b76f-6c9b-49fe-850b-a1034a63fd64.png">

- The statuses calculation was refactored in a less verbose way. 
- Fixed an error in which array_unique returned an associative version of the issues array sometimes. This was making Frontend part to fail because associative array is parsed as an object in JSON instead of an array. 
- Tweak: In case we find unknown statues or disabled global states we will hide the notice and the status.
- Fix: Added cooldown property instead of `nextAttempt` in the default state that was missed in the last PR. 


```
$response = [
				                 'programTypeA' => [
					                 'status' => 200,
					                 'data'   => [
						                 "globalState" =>  "ENABLED",
						                 'regionStatuses' => [
							                 [
								                 'reviewEligibilityStatus' => 'INELIGIBLE',
								                 'eligibilityStatus'       => 'APPROVED',
							                 ],
							                 [
								                 'reviewEligibilityStatus' => 'ELIGIBLE',
								                 'eligibilityStatus'       => 'DISAPPROVED',
								                 'reviewIssues'            => [ 'one' ]
							                 ],
						                 ]
					                 ]
				                 ],
				                 'programTypeB' => [
					                 'status' => 200,
					                 'data'   => [
						                 "globalState" =>  "ENABLED",
						                 'regionStatuses' => [
							                 [
								                 'reviewEligibilityStatus' => 'INELIGIBLE',
								                 'eligibilityStatus'       => 'WARNING',
								                 'reviewIssues'            => [ 'two' ]
							                 ],
							                 [
								                 'reviewEligibilityStatus' => 'ELIGIBLE',
								                 'eligibilityStatus'       => 'DISAPPROVED',
								                 'reviewIssues'            => [ 'one' ]
							                 ],
							                 [
								                 'reviewEligibilityStatus' => 'INELIGIBLE',
								                 'eligibilityStatus'       => 'UNDER_REVIEW'
							                 ],
						                 ]
					                 ]
				                 ]
			                 ];
```
### Detailed test instructions:
1. Checkout this issue and run `npm run dev` to generate the assets
2. In `src/API/Site/Controllers/MerchantCenter/RequestReviewController.php::get_review_read_callback()` you can hardcode the response with something like the provided above 
3. Test that if any globalState is NO_OFFERS_UPLOADED we get ONBOARDING status
4. Check that if any statues has ONBOARDING as eligibilityStatus we show ONBOARDING as status
5. Check that if any statues has DISAPPROVED as eligibilityStatus, and no ONBOARDING in any of them we show DISAPPROVED as status
6. Check that if any statues has WARNING as eligibilityStatus, and no ONBOARDING or DISAPPROVED in any of them we show WARNING as status
7. Check that if any statues has UNDER_REVIEW as eligibilityStatus, and no ONBOARDING or DISAPPROVED or WARNING in any of them we show UNDER_REVIEW as status
8. Check that if any statues has PENDING_REVIEW as eligibilityStatus, and no ONBOARDING or DISAPPROVED or WARNING or UNDER_REVIEW in any of them we show PENDING_REVIEW as status
9.  Check that if any statues has APPROVED as eligibilityStatus, and no ONBOARDING or DISAPPROVED or WARNING or UNDER_REVIEW or PENDING_REVIEW in any of them we show APPROVED as status
10. Rest of cases will show null as status. this could be tested by adding for example "globalState" =>  "UNKNOWN" in any of the programs inside the response.
